core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -706,8 +707,10 @@ public InodeDirectory getRoot() {
         newDir.setInternalAcl(pair.getFirst());
         newDir.setDefaultACL(pair.getSecond());
       }
+      String newDirPath = k == 0 ? ROOT_PATH
```

Comment:
The loop starts `k` from `pathIndex`. That doesn't have to start from `0`, right? If it starts from a non-zero index, does this still work?

Reply:
Good catch! Fixed.

Reply:
Should the creation of `pathBuilder` take care of the `k==0` case, and here we can just call `toString` on the builder?

Reply:
To simplify the logic I am keeping the `pathBuilder` free from the trailing slash. Therefore I have to special case the `k=0` situation where it has to output `/` but the `pathBuilder` cannot contains trailing slash.

Reply:
SG